respy/simulate.py

```diff
@@ -138,25 +138,39 @@ def _simulate_data(state_space, base_draws_sim, base_draws_wage, optim_paras, op
 
     base_draws_wage_transformed = np.exp(base_draws_wage * optim_paras["meas_error"])
 
+    # Create observables for simulation and store them in an extra container that
+    # will be added to the state space container later
+    container_obs = ()
+    for observable in optim_paras["observables"].keys():
+        container_obs += (
+            _get_random_initial_observable(observable, options, optim_paras),
+        )
+
+    # Create a DataFrame to match columns to covariates. Is changed in-place.
+    if container_obs != ():
+        states_df = pd.DataFrame(
+            np.column_stack(container_obs),
+            columns=[x for x in optim_paras["observables"].keys()],
+        ).assign(period=0)
+    else:
+        states_df = pd.DataFrame().assign(period=0)
+
     # Create initial experiences, lagged choices and types for agents in simulation.
     container = ()
     for choice in optim_paras["choices_w_exp"]:
-        container += (_get_random_initial_experience(choice, optim_paras, options),)
-
-    # Create a DataFrame to match columns to covariates. Is changed in-place.
-    states_df = pd.DataFrame(
-        np.column_stack(container),
-        columns=[f"exp_{i}" for i in optim_paras["choices_w_exp"]],
-    ).assign(period=0)
+        states_df[f"exp_{choice}"] = _get_random_initial_experience(
```

Comment:
You are completely right here. Delete the two containers and do everything with `states_df`. No in-place changes anymore. Pass `states_df` to the sampling functions and return the df again.